test/NuGet.Clients.Tests/NuGet.CommandLine.Test/NuGetUpdateCommandTests.cs

```diff
@@ -1537,5 +1537,122 @@ public async Task UpdateCommand_Native_JS_Projects_Success()
                 Assert.True(r.Item1 == 0, "Output is " + r.Item2 + ". Error is " + r.Item3);
             }
         }
+
+        [Fact]
+        public async Task UpdateCommand_DependencyResolution_Success()
+        {
+            // Test variations
+            // dependencyVersion, expected version
+            string[,] variations =
```

Comment:
Use a Theory and providing the values there. 
Otherwise this provides a moving target if the tests ever fail. (you don't want a fail fast behavior here).